tensorflow/contrib/seq2seq/python/ops/layers.py

```diff
@@ -20,15 +20,185 @@
 from __future__ import division
 from __future__ import print_function
 
+from tensorflow.python.ops import tensor_array_ops
 from tensorflow.python.ops import array_ops
+from tensorflow.python.ops import variable_scope as vs
+from tensorflow.python.ops import math_ops
+from tensorflow.python.ops import control_flow_ops
+from tensorflow.python.ops import rnn
 
+from tensorflow.contrib.layers import fully_connected
+
+from functools import partial
 
 __all__ = ["rnn_decoder",
            "rnn_decoder_attention"]
 
+"""Used to project encoder state in `rnn_decoder`"""
+encoder_projection = partial(fully_connected, activation_fn=math_ops.tanh)
 
-def rnn_decoder(*args, **kwargs):
-  pass
+
+def rnn_decoder(cell, decoder_inputs, initial_state,
+                sequence_length, decoder_fn,
+                encoder_projection=encoder_projection,
+                parallel_iterations=None, swap_memory=False,
```

Comment:
"stolen" from the raw_rnn [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/rnn.py#L1033) and [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/rnn.py#L1158).

Perhaps we could write something in the docstring saying "(Default: None) ... will get set to 32 ny 'raw_rnn' if set to None"?